gpflow/conditionals/util.py

```diff
@@ -51,6 +51,16 @@ def base_conditional(Kmn: tf.Tensor,
         0)  # [N]
     Kmn = tf.transpose(Kmn, perm)  # [..., M, N]
 
+    shape_constraints = [
+        (Kmn, [..., 'M', 'N']),
+        (Kmm, ['M', 'M']),
+        (Knn, [..., 'N', 'N'] if full_cov else [..., 'N']),
+        (f, ['M', 'R']),
+    ]
+    if q_sqrt is not None:
+        shape_constraints.append((q_sqrt, ['R', 'M', 'M']))
+    tf.debugging.assert_shapes(shape_constraints)
```

Comment:
This is super cool and worth of making it as a separate internal module function.

Reply:
Thanks to GPflow user @bodin-e (tensorflow#25422)... :)
What do you mean by making it an internal function?

Reply:
I wonder if it were more readable with tf.debugging.assert_shapes per each shape instead of having a list.

Reply:
Then you can't cross-check that the dimensions match each other (e.g. that the "M" of Kmn is the same as the "M" of Kmm).